### DISCUSSION TITLE
How do I use ports as runtime dynamic libraries?

### DISCUSSION THREAD

**Discussion by Hex4dec** [emscripten-core/emscripten#21962] [Q&A] [answered]
I'm porting OpenJDK to Emscripten, and as Java programs can call natives with runtime dynamic linking, it would be nice to get some libraries like Regal, GLFW, and so on and so forth working.

Is cloning the ports I require and then building them as side modules the only approach? Is there some magic I can do with `--use-port`?

**Comment by Hex4dec** [comment, marked as answer]
I've tried `emcc -sSIDE_MODULE --use-port=regal null.c -o out/regal.wasm` where `null.c` is an empty C source file. That doesn't work when linked to with `dlopen()`.

Seems like `emcc -sSIDE_MODULE=1 .../emsdk/upstream/emscripten/cache/sysroot/lib/wasm32-emscripten/pic/libregal.a -o regal.wasm` works. Good enough!

**Reply by sbc100** [reply]
You should be able to do `-lregal` instead of `.../emsdk/upstream/emscripten/cache/sysroot/lib/wasm32-emscripten/pic/libregal.a`
